diff --git a/cmake/opentelemetry-proto.cmake b/cmake/opentelemetry-proto.cmake
index 37d45da..f5d4759 100644
--- a/cmake/opentelemetry-proto.cmake
+++ b/cmake/opentelemetry-proto.cmake
@@ -189,8 +189,9 @@ endif()
 include_directories("${GENERATED_PROTOBUF_PATH}")
 
 if(WITH_OTLP_GRPC)
+  find_library(grpc++ grpc++)
   add_library(
-    opentelemetry_proto STATIC
+    opentelemetry_proto SHARED
     ${COMMON_PB_CPP_FILE}
     ${RESOURCE_PB_CPP_FILE}
     ${TRACE_PB_CPP_FILE}
@@ -202,9 +203,10 @@ if(WITH_OTLP_GRPC)
     ${LOGS_SERVICE_GRPC_PB_CPP_FILE}
     ${METRICS_SERVICE_PB_CPP_FILE}
     ${METRICS_SERVICE_GRPC_PB_CPP_FILE})
+  target_link_libraries(opentelemetry_proto PUBLIC grpc++)
 else()
   add_library(
-    opentelemetry_proto STATIC
+    opentelemetry_proto SHARED
     ${COMMON_PB_CPP_FILE}
     ${RESOURCE_PB_CPP_FILE}
     ${TRACE_PB_CPP_FILE}
diff --git a/exporters/jaeger/CMakeLists.txt b/exporters/jaeger/CMakeLists.txt
index 017b43d..82ce6a6 100644
--- a/exporters/jaeger/CMakeLists.txt
+++ b/exporters/jaeger/CMakeLists.txt
@@ -32,8 +32,7 @@ target_include_directories(
 
 target_link_libraries(
   opentelemetry_exporter_jaeger_trace
-  PUBLIC opentelemetry_resources opentelemetry_http_client_curl
-  PRIVATE thrift::thrift)
+  PUBLIC opentelemetry_resources opentelemetry_http_client_curl thrift::thrift)
 
 if(MSVC)
   target_compile_definitions(opentelemetry_exporter_jaeger_trace
