diff --git a/exporters/otlp/CMakeLists.txt b/exporters/otlp/CMakeLists.txt
index c822859..7f9fbca 100644
--- a/exporters/otlp/CMakeLists.txt
+++ b/exporters/otlp/CMakeLists.txt
@@ -137,9 +137,10 @@ install(
   PATTERN "*.h")
 
 if(BUILD_TESTING)
+  find_library(gRPC grpc)
   add_executable(otlp_recordable_test test/otlp_recordable_test.cc)
-  target_link_libraries(otlp_recordable_test ${GTEST_BOTH_LIBRARIES}
-                        ${CMAKE_THREAD_LIBS_INIT} opentelemetry_otlp_recordable)
+  target_link_libraries(otlp_recordable_test PUBLIC ${GTEST_BOTH_LIBRARIES}
+                        ${CMAKE_THREAD_LIBS_INIT} opentelemetry_otlp_recordable grpc)
   gtest_add_tests(
     TARGET otlp_recordable_test
     TEST_PREFIX exporter.otlp.
