diff --git a/tests/test_functional.py b/tests/test_functional.py
index 77cce9b..3110ba2 100644
--- a/tests/test_functional.py
+++ b/tests/test_functional.py
@@ -369,7 +369,7 @@ def test_buildlatex_with_align():
     assert (re.search(br'\\begin\{figure\}\[htbp\]\\begin\{flushright\}', out)
             or re.search(br'\\begin\{wrapfigure\}\{r\}', out))
 
-
+@unittest.skip("Further investigation is needed")
 @with_runsphinx('pdf')
 def test_buildpdf_simple():
     """Generate simple PDF
